0.11.2: pytest is failing

Looks like pytest is failing in `tests/test_unittest.py::TestUnitTestIntegration::test_failed_test_case` unit
```console
+ PYTHONPATH=/home/tkloczko/rpmbuild/BUILDROOT/python-flexmock-0.11.2-2.fc35.x86_64/usr/lib64/python3.8/site-packages:/home/tkloczko/rpmbuild/BUILDROOT/python-flexmock-0.11.2-2.fc35.x86_64/usr/lib/python3.8/site-packages
+ /usr/bin/pytest -ra --ignore tests/test_teamcity.py
=========================================================================== test session starts ============================================================================
platform linux -- Python 3.8.12, pytest-6.2.5, py-1.11.0, pluggy-1.0.0
rootdir: /home/tkloczko/rpmbuild/BUILD/flexmock-0.11.2
collected 681 items

tests/test_doctest.py ..                                                                                                                                             [  0%]
tests/test_flexmock.py ............................................................................................................................................. [ 20%]
........................................................................................                                                                             [ 33%]
tests/test_pytest.py ............................................................................................................................................... [ 54%]
.................................................................................................................................................................... [ 79%]
.....................................................................................................................................xxxx                            [ 99%]
tests/test_testtools.py .....                                                                                                                                        [ 99%]
tests/test_unittest.py F                                                                                                                                             [100%]

================================================================================= FAILURES =================================================================================
______________________________________________________________ TestUnitTestIntegration.test_failed_test_case _______________________________________________________________

    def flexmock_teardown() -> None:
        """Performs flexmock-specific teardown tasks."""
        saved = {}
        instances = []
        classes = []
        for mock_object, expectations in FlexmockContainer.flexmock_objects.items():
            saved[mock_object] = expectations[:]
            for expectation in expectations:
                expectation._reset()
            for expectation in expectations:
                # Remove method type attributes set by flexmock. This needs to be done after
                # resetting all the expectations because method type is needed in expectation teardown.
                if inspect.isclass(mock_object) or hasattr(mock_object, "__class__"):
                    try:
                        delattr(mock_object._object, f"{expectation._name}__flexmock__method_type")
                    except (AttributeError, TypeError):
                        pass
        for mock in saved:
            obj = mock._object
            if not isinstance(obj, Mock) and not inspect.isclass(obj):
                instances.append(obj)
            if inspect.isclass(obj):
                classes.append(obj)
        for obj in instances + classes:
            for attr in UPDATED_ATTRS:
                try:
                    obj_dict = obj.__dict__
                    if obj_dict[attr].__code__ is Mock.__dict__[attr].__code__:
                        del obj_dict[attr]
                except Exception:
                    try:
                        if getattr(obj, attr).__code__ is Mock.__dict__[attr].__code__:
                            delattr(obj, attr)
                    except AttributeError:
                        pass
        FlexmockContainer.teardown_properties()
        FlexmockContainer.reset()

        # make sure this is done last to keep exceptions here from breaking
        # any of the previous steps that cleanup all the changes
        for mock_object, expectations in saved.items():
            for expectation in expectations:
>               expectation._verify()

../../BUILDROOT/python-flexmock-0.11.2-2.fc35.x86_64/usr/lib/python3.8/site-packages/flexmock/_api.py:569:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
../../BUILDROOT/python-flexmock-0.11.2-2.fc35.x86_64/usr/lib/python3.8/site-packages/flexmock/_api.py:1226: in _verify
    self.__raise(
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <flexmock._api.Expectation object at 0x7fba265869d0>, exception = <class 'flexmock.exceptions.MethodCallError'>
message = 'this_test_should_fail() expected to be called exactly 1 time, called 0 times'

    def __raise(self, exception: Type[BaseException], message: str) -> NoReturn:
        """Safe internal raise implementation.

        In case we're patching builtins, it's important to reset the
        expectation before raising any exceptions or else things like
        open() might be stubbed out and the resulting runner errors are very
        difficult to diagnose.
        """
        self._reset()
>       raise exception(message)
E       flexmock.exceptions.MethodCallError: this_test_should_fail() expected to be called exactly 1 time, called 0 times

../../BUILDROOT/python-flexmock-0.11.2-2.fc35.x86_64/usr/lib/python3.8/site-packages/flexmock/_api.py:759: MethodCallError
========================================================================= short test summary info ==========================================================================
XFAIL tests/test_pytest.py::TestFailureOnException::test_exception
XFAIL tests/test_pytest.py::TestFailureOnException::test_flexmock_raises_if_no_exception
XFAIL tests/test_pytest.py::TestFailureOnException::test_flexmock_doesnt_override_existing_exception
XFAIL tests/test_pytest.py::TestFailureOnException::test_flexmock_doesnt_override_assertion
FAILED tests/test_unittest.py::TestUnitTestIntegration::test_failed_test_case - flexmock.exceptions.MethodCallError: this_test_should_fail() expected to be called exactl...
================================================================= 1 failed, 676 passed, 4 xfailed in 2.27s =================================================================
```

--- a/.github/workflows/coverage.yml
+++ b/.github/workflows/coverage.yml
@@ -33,7 +33,7 @@ jobs:
           poetry run coverage run --parallel-mode --source flexmock -m unittest tests/test_flexmock.py
           poetry run coverage run --parallel-mode --source flexmock tests/test_unittest.py
           poetry run coverage run --parallel-mode --source flexmock tests/test_doctest.py
-          poetry run coverage run --parallel-mode --source flexmock -m pytest tests/test_pytest.py
+          poetry run coverage run --parallel-mode --source flexmock -m pytest
           poetry run coverage combine
           poetry run coverage xml
 

--- a/Makefile
+++ b/Makefile
@@ -30,7 +30,7 @@ pytest:
 	@printf '\n\n*****************\n'
 	@printf '$(color)Running pytest$(off)\n'
 	@printf '*****************\n'
-	pytest tests/test_pytest.py
+	pytest
 
 .PHONY: unittest
 unittest:

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -117,3 +117,7 @@ no_implicit_optional = true
 [[tool.mypy.overrides]]
 module = "tests.*"
 ignore_errors = true
+
+[tool.pytest.ini_options]
+python_files = ["test_pytest.py"]
+xfail_strict = true

